enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,315 @@
+	ModerateCount int `json:"moderate"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total"`
+}
+
+type GathererStatus struct {
+	// Name is the name of the gatherer.
+	Name string `json:"name"`
+	// GathererConditions provide details on the status of each gatherer.
+	GathererConditions []GathererCondition `json:"conditions"`
+	// DurationMillisecond represents the time spent gathering.
+	DurationMillisecond int64 `json:"durationMillisecond"`
+	// RecordsCount represents the number of records produced by the gatherer.
+	RecordsCount int `json:"recordsCount"`
+}
+
+type GathererCondition struct {
+	// Type of the gatherer condition
+	Type GathererConditionType `json:"type"`
+	// Status is last known status of the particular gatherer condition
+	Status corev1.ConditionStatus `json:"status"`
+	// Messages is an optional attribute that provides error and warning messages
+	// from the gatherer
+	Messages []string `json:"messages,omitempty"`
+}
+
+// GathererConditionType is a valid value for GathererCondition.Type
+type GathererConditionType string
+
+const (
+	// Gatherer was successful withnout any error reported
+	Successful GathererConditionType = "Successful"
+	// Gatherer was disabled by user
+	Disabled GathererConditionType = "Disabled"
+	// Gatherer failed to run
+	Failed GathererConditionType = "Failed"
+	// Gatherer was running, but there were some errors
+	Warning GathererConditionType = "Warning"
+	// Gatherer panicked
+	Panic GathererConditionType = "Panic"
+)
+
+const (
+	// There is no gathering in progress
+	GatherStatusWaiting GatherStatus = "Waiting"
+	// There is running gathering in progress
+	GatherStatusRunning GatherStatus = "Running"
+)
+
+// GatherStatus is a valid value for GatheringStatus.Status
+type GatherStatus string
+```
+
+### User Stories
+#### Force gathering - refresh Insights status
+As an OCP user/cluster-admin I would like to be able to trigger on demand gathering (there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status). Steps:
+- the user will set a new value of `ForceGatheringReason`
+- if all the gatherers are disabled (`disabledGatherers: ["all"]`), nothing happens
+- if there is some gathering in progress (`Status` is "running" in the `GatheringStatus` section) then it will be interrupted and the new gathering will be triggered
+- if there is not any gathering in progress (`Status` is "waiting" in the `GatheringStatus` section) then the new gathering will be triggered almost immediately
+- user can check the `Status` or the `LastGatheringTime` attribute in the `GatheringStatus` section to check if the new gathering already finished
```

Comment:
Same question as above, what happens with `ForceGatheringReason` field after a successful run?

Reply:
I compared this with `ForceRedeploymentReason` in our operators, and they leave it set, so for consistency I'm fine leaving it set. You'll easily know when the field changes to initiate a new run from informers.